Error: Could not load rockspec file /apisix/./rockspec/apisix-master-0.rockspec (Error loading file: [string "/apisix/./rockspec/apisix-master-0.rockspec"]:77: '}' expected (to close '{' at line 33) near '.') The command '/bin/sh -c /install-common.sh install_apisix     && /determine-dist.sh' returned a non-zero code: 1 make: *** [Makefile:121: build-apisix-rpm] Error 1

cause by sed 
```bash
https://github.com/api7/apisix-build-tools/blob/master/utils/install-common.sh#L77
```

Error build action
> https://github.com/api7/apisix-build-tools/runs/5429589069?check_suite_focus=true
```bash
Error: Could not load rockspec file /apisix/./rockspec/apisix-master-0.rockspec (Error loading file: [string "/apisix/./rockspec/apisix-master-0.rockspec"]:77: '}' expected (to close '{' at line 33) near '.')
The command '/bin/sh -c /install-common.sh install_apisix     && /determine-dist.sh' returned a non-zero code: 1
make: *** [Makefile:121: build-apisix-rpm] Error 1
```

> for https://github.com/apache/apisix/blob/master/rockspec/apisix-master-0.rockspec Added 

```bash
dependencies = {
    "net-url = 0.9-1",
}
```

replaced with sed result
```bash
dependencies = {
    "net-url = "./apisix",
}
```
I think this https://github.com/api7/apisix-build-tools/blob/master/utils/install-common.sh#L77 sed maybe need update

--- a/.github/workflows/package-apisix-dashboard-rpm-el8.yml
+++ b/.github/workflows/package-apisix-dashboard-rpm-el8.yml
@@ -49,6 +49,12 @@ jobs:
 
       - name: Install rpm package
         run: |
+          docker exec centos8Instance bash -c "sed -i 's/mirrorlist/#mirrorlist/g' /etc/yum.repos.d/CentOS-Linux-*"
+          docker exec centos8Instance bash -c "sed -i 's|#baseurl=http://mirror.centos.org|baseurl=http://vault.centos.org|g' /etc/yum.repos.d/CentOS-Linux-*"
+          docker exec centos8Instance bash -c "dnf install -y centos-release-stream"
+          docker exec centos8Instance bash -c "dnf swap -y centos-{linux,stream}-repos"
+          docker exec centos8Instance bash -c "dnf distro-sync -y"
+          
           docker exec centos8Instance bash -c "yum install -y /apisix-dashboard/apisix-dashboard-${DASHBOARD_VERSION}-0.el8.x86_64.rpm"
           docker logs centos8Instance
           docker exec centos8Instance bash -c "cd /usr/local/apisix/dashboard/ && nohup ./manager-api &"

--- a/utils/install-common.sh
+++ b/utils/install-common.sh
@@ -2,6 +2,17 @@
 set -euo pipefail
 set -x
 
+patch_centos8_repo() {
+    # switch yum repo source
+    sed -i 's/mirrorlist/#mirrorlist/g' /etc/yum.repos.d/CentOS-Linux-*
+    sed -i 's|#baseurl=http://mirror.centos.org|baseurl=http://vault.centos.org|g' /etc/yum.repos.d/CentOS-Linux-*
+
+    # rebuild repo cache
+    dnf install -y centos-release-stream
+    dnf swap -y centos-{linux,stream}-repos
+    dnf distro-sync -y
+}
+
 install_apisix_dependencies_deb() {
     install_dependencies_deb
     install_openresty_deb
@@ -9,12 +20,16 @@ install_apisix_dependencies_deb() {
 }
 
 install_apisix_dependencies_rpm() {
+    patch_centos8_repo
+
     install_dependencies_rpm
     install_openresty_rpm
     install_luarocks
 }
 
 install_dependencies_rpm() {
+    patch_centos8_repo
+
     # install basic dependencies
     yum -y install wget tar gcc automake autoconf libtool make curl git which unzip sudo
     yum -y install epel-release
@@ -110,6 +125,8 @@ install_golang() {
 }
 
 install_dashboard_dependencies_rpm() {
+    patch_centos8_repo
+
     yum install -y wget curl git which gcc make
     curl --silent --location https://dl.yarnpkg.com/rpm/yarn.repo | tee /etc/yum.repos.d/yarn.repo
     sh -c "$(curl -fsSL https://rpm.nodesource.com/setup_14.x)"

--- a/utils/install-common.sh
+++ b/utils/install-common.sh
@@ -3,6 +3,9 @@ set -euo pipefail
 set -x
 
 patch_centos8_repo() {
+    if [[ $(rpm --eval '%{centos_ver}') != "8" ]]; then
+        return
+    fi
     # switch yum repo source
     sed -i 's/mirrorlist/#mirrorlist/g' /etc/yum.repos.d/CentOS-Linux-*
     sed -i 's|#baseurl=http://mirror.centos.org|baseurl=http://vault.centos.org|g' /etc/yum.repos.d/CentOS-Linux-*

--- a/utils/install-common.sh
+++ b/utils/install-common.sh
@@ -87,13 +87,8 @@ is_newer_version() {
 }
 
 install_apisix() {
-    # show awk version
-    awk --version
     mkdir -p /tmp/build/output/apisix/usr/bin/
     cd /apisix
-    # remove useless code for build
-    sed -i 's/url.*/url = ".\/apisix",/' rockspec/apisix-master-${iteration}.rockspec
-    sed -i 's/branch.*//' rockspec/apisix-master-${iteration}.rockspec
     # build the lib and specify the storage path of the package installed
     luarocks make ./rockspec/apisix-master-${iteration}.rockspec --tree=/tmp/build/output/apisix/usr/local/apisix/deps --local
     chown -R "$(whoami)":"$(whoami)" /tmp/build/output

